src/cargo/core/compiler/unit_dependencies.rs

```diff
@@ -281,10 +281,27 @@ fn compute_deps(
         let start = ret.len();
         if state.config.cli_unstable().dual_proc_macros && lib.proc_macro() && !unit.kind.is_host()
         {
-            let unit_dep = new_unit_dep(state, unit, pkg, lib, dep_unit_for, unit.kind, mode)?;
+            let unit_dep = new_unit_dep(
+                state,
+                unit,
+                pkg,
+                lib,
+                dep_unit_for,
+                unit.kind,
+                mode,
+                /*artifact*/ false,
```

Comment:
I couldn't agree more. I think I saw this style originally [here](https://github.com/rust-lang/cargo/blob/9bae0197dcacc1ae480c3b4d756375821f77d5c9/src/cargo/ops/cargo_compile.rs#L1054-L1056) and adopted it.

Knowing RFC-3176 and before making the change, let me add that in the mentioned PR this changes to `Option<&Artifact>`. When it's `None`, it's still documented like `/*artifact*/ None`. Now I am hesitant to add something like `IsArtifact::Yes/No` as it will soon turn into `Artifact::None/Given(&Artifact)` which essentially is a named custom `Option` which I thought was discouraged.

The code itself usually matches on it anyway so won't benefit from having an `Option` and its combinator methods. The resulting question is: Should we introduce the enum pattern knowing that it will turn into an option-replacing pattern in RFC-3176 which might be unwanted?

Thanks for your advice.

Reply:
@Byron I do think `IsArtifact` would make this PR more readable. I'd suggest going ahead and making that change. I do think in the RFC-3176 implementation you should just use `Option` and `None`, rather than recreating an `Option`-like type.

Reply:
Thanks for the clarification. It's now implemented here: 76cd48a2d62d74e043a1a482199c5bb920f50311 , and definitely nicer to look at.

(It's a bit sad that RFC-3176 has to turn these `IsArtifact::No` back into `None`. Maybe I should then create a constant that assigns a name to `None`, such as `const NO_ARTIFACT: Option<Artifact> = None`.)